android/ExoPlayer-pre2.10/app/src/main/java/io/streamroot/dna/samples/exoplayer/ExoPlayerInteractor.kt

```diff
@@ -0,0 +1,122 @@
+package io.streamroot.dna.samples.exoplayer
+
+import android.os.Looper
+import com.google.android.exoplayer2.ExoPlayer
+import com.google.android.exoplayer2.LoadControl
+import com.google.android.exoplayer2.Timeline
+import io.streamroot.dna.core.PlayerInteractor
+import io.streamroot.dna.core.TimeRange
+import java.util.concurrent.TimeUnit
+
+private interface BufferTargetBridge {
+    fun bufferTarget() : Double = 0.0
+    fun setBufferTarget(bufferTarget: Double) {}
+}
+
+private class BufferTargetBridgeDefault : BufferTargetBridge
+
+private abstract class LoadControlBufferTargetBridge(protected val loadControl: LoadControl) : BufferTargetBridge {
+
+    protected fun LoadControl.getAccessibleFieldElseThrow(fieldName: String) = runCatching {
+        val minBufferField = this::class.java.getDeclaredField(fieldName)
+        minBufferField.isAccessible = true
+        minBufferField
+    }.getOrNull() ?: throw IllegalArgumentException("Impossible to retrieve field `$fieldName` value from LoadControl of type `${this::class.java.simpleName}`")
+
+    protected fun LoadControl.getLongFromFieldElseThrow(fieldName: String) = runCatching {
+        getAccessibleFieldElseThrow(fieldName).getLong(this)
+    }.getOrNull() ?: throw IllegalArgumentException("Impossible to retrieve long `$fieldName` value from LoadControl of type `${this::class.java.simpleName}`")
+
+    companion object {
+        private const val MAX_BUFFER_FIELD_NAME = "maxBufferUs"
+    }
+
+    protected val maxBufferField = loadControl.getAccessibleFieldElseThrow(MAX_BUFFER_FIELD_NAME)
+    protected abstract val minBufferUs: Long
+
+    override fun bufferTarget(): Double {
+        return runCatching {
+            maxBufferField.getLong(loadControl).let { TimeUnit.MICROSECONDS.toSeconds(it) }.toDouble()
+        }.getOrNull() ?: super.bufferTarget()
+    }
+
+    override fun setBufferTarget(bufferTarget: Double) {
+        val maxBufferUs = TimeUnit.SECONDS.toMicros(bufferTarget.toLong())
+        if (maxBufferUs > minBufferUs) runCatching {
+            maxBufferField.setLong(
+                    loadControl,
+                    maxBufferUs
+            )
+        }
+    }
+}
+
+private class LoadControlBufferTargetBridgeV1(loadControl: LoadControl)
+    : LoadControlBufferTargetBridge(loadControl) {
+    companion object {
+        private const val MIN_BUFFER_FIELD_NAME = "minBufferUs"
+    }
+
+    override val minBufferUs = loadControl.getLongFromFieldElseThrow(MIN_BUFFER_FIELD_NAME)
+}
```

Comment:
@tri170391, since this project is garanteed 2.11+, should we remove the bridge 1 ?

Reply:
Well that would complicate the docs though @emilepommier

Reply:
Decided to remove Bridge V1 for 2.11+ :)